### PR TITLE
Enabled 3.6, 3.7, 3.8 to conda build automation, removed date from release versioning, updated badged.

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
Added head/headless options, enabled 3.6, 3.7, 3.8 to conda build automation

Regarding badges, propose to avoid deduplication and remove some of them: 
- Removed black badge as pre-commit badge is presented and if pre-commit installed black will be picked up, badge is important for contributors only
- Removed version release badge as version is already mentioned on Anaconda release and should be the same
- Added Docs badge as we want to focus users more that Documentation is available and it's alive
- Added python languages supported.  

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by dhruvbatra — APPROVED** [review]

File: conda-build/Dockerfile [hunk 1]
```diff
@@ -35,6 +34,7 @@ RUN bash ./install_conda.sh && rm install_conda.sh
 
 RUN anaconda login --username $AIHABITAT_CONDA_CHN --password $AIHABITAT_CONDA_CHN_PWD
 RUN conda create --name py36 python=3.6 -y && conda init bash
+RUN conda create --name py37 python=3.7 -y
```
Comment (erikwijmans) [inline conda-build/Dockerfile:37]: I don't believe this is needed.  `condo-build` takes are of setting up a py3.7 environment when you build a py3.7 package.

Comment (mathfac) [reply conda-build/Dockerfile:37]: @erikwijmans, good catch. Noticed that as well.

Comment (Skylion007) [reply conda-build/Dockerfile]: Can we also remove the python=3.6 as well?

**Review by Skylion007 — CHANGES_REQUESTED** [review]
Why not enable builds for Python 3.8 and 3.9 as well?

Also we should update the badge in the ReadMe to show we support multiple Python versions.

**Comment by Skylion007** [comment]
We probably also need to update the build matrix then to do to install_and_test_ubuntu for 3.7, right? (in parallel)

File: conda-build/Dockerfile (hunk 1 above)
Comment (mathfac) [reply conda-build/Dockerfile]: No, we can't. That's root environment that we use to run conda build that has all dependency installed.

**Comment by mathfac** [comment]
Adding `3.8` is good idea based on statistics from [Pytorch Conda repo](https://anaconda.org/pytorch/pytorch/files?sort=time&sort_order=desc), `3.9` maybe too early.
Regarding running testing for 3.7. I would rather add some integration tests for conda package builded than running all over tests on one more setup. Let's make conda packages testing separate PR.

**Comment by mathfac** [comment]
Regarding badges, propose to avoid deduplication and remove some of them: 
- Removed black badge as pre-commit badge is presented and if pre-commit installed black will be picked up, badge is important for contributors only
- Removed version release badge as version is already mentioned on Anaconda release and should be the same
- Added Docs badge as we want to focus users more that Documentation is available and it's alive
- Added python languages supported.

**Comment by Skylion007** [comment]
>> Removed black badge as pre-commit badge is presented and if pre-commit installed black will be picked up
How I wish I could actually enforce pre-commit being installed properly, but there are contributors who don't run it on every commit.

File: README.md [hunk 2]
```diff
@@ -3,10 +3,9 @@
 [![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/facebookresearch/habitat-sim/blob/master/LICENSE)
 [![Anaconda-Server Badge](https://anaconda.org/aihabitat/habitat-sim/badges/version.svg)](https://anaconda.org/aihabitat/habitat-sim)
 [![Anaconda-Server Badge](https://anaconda.org/aihabitat/habitat-sim/badges/platforms.svg)](https://anaconda.org/aihabitat/habitat-sim)
+[![Documentation](https://img.shields.io/badge/docs-automated-green.svg)](https://aihabitat.org/docs/habitat-sim/)
 [![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit&logoColor=white)](https://github.com/pre-commit/pre-commit)
-[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
-[![GitHub release (latest by date)](https://img.shields.io/github/v/release/facebookresearch/habitat-sim)](https://github.com/facebookresearch/habitat-sim/releases/latest)
```
Comment (Skylion007) [inline README.md:8]: Strongly disagree with as the Anaconda Release version looks entirely different from the semantic versioning we actually using in setup.py and elsewhere.

File: README.md [hunk 3]
```diff
@@ -3,10 +3,9 @@
...
-[![Python 3.6](https://img.shields.io/badge/python-3.6-blue.svg)](https://www.python.org/downloads/release/python-360/)
+[![Python 3.6](https://img.shields.io/badge/python-3.6%20%7C%203.7%20%7C%203.8-blue.svg)](https://www.python.org/downloads/release/)
```
Comment (Skylion007) [inline README.md:8]: Change the badge name here as well.

File: README.md [hunk 4]
```diff
@@ -3,10 +3,9 @@
 [![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/facebookresearch/habitat-sim/blob/master/LICENSE)
 [![Anaconda-Server Badge](https://anaconda.org/aihabitat/habitat-sim/badges/version.svg)](https://anaconda.org/aihabitat/habitat-sim)
 [![Anaconda-Server Badge](https://anaconda.org/aihabitat/habitat-sim/badges/platforms.svg)](https://anaconda.org/aihabitat/habitat-sim)
+[![Documentation](https://img.shields.io/badge/docs-automated-green.svg)](https://aihabitat.org/docs/habitat-sim/)
```
Comment (Skylion007) [inline README.md:6]: This should actually be a circleci badge that points to our automatic docs deployment job so we know if docs suddenly start failing.

**Review by Skylion007 — CHANGES_REQUESTED** [review]
Issues with some of the badge stuff

File: README.md (hunk 4 above)
Comment (mathfac) [reply README.md:6]: @Skylion007  thank you for the suggestion: I can't find [CircleCI tags](https://circleci.com/docs/2.0/status-badges/) or Shields.io supporting specific step in workflow status represented as the badge. If our CircleCI build badge is green => Docs build is green for sure.

File: README.md (hunk 2 above)
Comment (mathfac) [reply README.md:8]: Can you elaborate more, here is the screen showing the versioning, here badge from Anaconda was cached by Github and will update:
Screen Shot 2020-09-18 at 10 45 28 AM
How different the versions are?

**Comment by mathfac** [comment]
> How I wish I could actually enforce pre-commit being installed properly, but there are contributors who don't run it on every commit.

Then they will get CI failed as soon as they send PR and discover that.

**Comment by mathfac** [comment]
Badges now:
Screen Shot 2020-09-18 at 10 49 10 AM

**Comment by mathfac** [comment]
[Slate Anaconda badges is known issue](https://github.com/ContinuumIO/anaconda-issues/issues/10231), switch to shields.io badge instead.

**Comment by mathfac** [comment]
Removed the date from versioning for releases, while left it for nightly builds.

File: README.md (hunk 4 above)
Comment (Skylion007) [reply README.md:6]: Yeah you are right, that's a huge oversight. It shouldn't be automated green then, it should informative blue.

**Review by Skylion007 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mathfac) [reply README.md:8]: Discussed with Aaron separately, removed the date from release versioning.

**Comment by mathfac** [comment]
9 packages produced. PR is ready to be merged.

**Pull request merged**